### PR TITLE
SAAS-20313/Front-Door-Waf-Enabled

### PR DESCRIPTION
**Pull request by alphadev4**

### CODE REVIEW

File: plugins/azure/frontdoor/frontDoorWafEnabled.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+            }
+            if (!profiles.data.length) {
+                helpers.addResult(results, 0, 'No existing Front Door profiles found', location);
+                return rcb();
+            }
+
+            var frontDoorPremium = false;
+
+            profiles.data.forEach(function(profile) {
+                if (!profile.id || !profile.sku || profile.sku.name.toLowerCase() != 'premium_azurefrontdoor') return;
```
Comment (alphadev4) [inline plugins/azure/frontdoor/frontDoorWafEnabled.js:37]:
```suggestion
                if (!profile.id) return;
```

File: plugins/azure/frontdoor/frontDoorWafEnabled.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+                        helpers.addResult(results, 2, 'Front Door profile does not have WAF enabled', location, profile.id);
+                    } else {
+                        helpers.addResult(results, 0, 'Front Door profile has WAF enabled', location, profile.id);
+                    }
+                }
+            });
+            
+            if (!frontDoorPremium) {
+                helpers.addResult(results, 0, 'No existing Front Door profiles found', location);
+            }
```
Comment (alphadev4) [inline plugins/azure/frontdoor/frontDoorWafEnabled.js:55]:
```suggestion
```

File: plugins/azure/frontdoor/frontDoorWafEnabled.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+                helpers.addResult(results, 0, 'No existing Front Door profiles found', location);
+                return rcb();
+            }
+
+            var frontDoorPremium = false;
+
+            profiles.data.forEach(function(profile) {
+                if (!profile.id || !profile.sku || profile.sku.name.toLowerCase() != 'premium_azurefrontdoor') return;
+                
+                frontDoorPremium = true;
```
Comment (alphadev4) [inline plugins/azure/frontdoor/frontDoorWafEnabled.js:39]:
```suggestion
```

File: plugins/azure/frontdoor/frontDoorWafEnabled.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+                helpers.addResult(results, 0, 'No existing Front Door profiles found', location);
+                return rcb();
+            }
+
+
+            profiles.data.forEach(function(profile) {
+                if (!profile.id) return;
+                
+                const afdSecurityPolicies = helpers.addSource(cache, source,
+                    ['afdSecurityPolicies', 'listByProfile', location, profile.id]);
```
Comment (alphadev4) [inline plugins/azure/frontdoor/frontDoorWafEnabled.js:39]:
```suggestion
                    ['afdSecurityPolicies', 'listByProfile', location, profile.id]);
                    
```

File: helpers/azure/api.js [hunk 5]
```diff
@@ -910,6 +910,14 @@ var postcalls = {
             properties: ['id'],
             url: 'https://management.azure.com/{id}/configurations?api-version=2017-12-01'
         }
+    },
+    afdSecurityPolicies: {
+        listByProfile: {
+            reliesOnPath: 'profiles.listPostgres',
```
Comment (mehakseedat63) [inline helpers/azure/api.js:916]:
```suggestion
            reliesOnPath: 'profiles.listPostgres',
```

Shouldn't this be 'profiles.list'?

**Pull request merged**